tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -5010,3 +5010,306 @@ TEST_F(EnergyPlusFixture, HPWH_Pumped_Stratified_Simultaneous)
+        "  ,                        !- Use Side Outlet Node Name",
+        "  ,                        !- Use Side Effectiveness",
+        "  ,                        !- Use Side Inlet Height {m}",
+        "  ,                        !- Use Side Outlet Height {m}",
+        "  ,                        !- Source Side Inlet Node Name",
+        "  ,                        !- Source Side Outlet Node Name",
+        "  ,                        !- Source Side Effectiveness",
+        "  ,                        !- Source Side Inlet Height {m}",
+        "  ,                        !- Source Side Outlet Height {m}",
+        "  FIXED,                   !- Inlet Mode",
+        "  ,                        !- Use Side Design Flow Rate {m3/s}",
+        "  ,                        !- Source Side Design Flow Rate {m3/s}",
+        "  ,                        !- Indirect Water Heating Recovery Time {hr}",
+        "  6,                       !- Number of Nodes",
+        "  ,                        !- Additional Destratification Conductivity {W/m-K}",
+        "  1,                       !- Node 1 Additional Loss Coefficient {W/K}",
+        "  ,                        !- Node 2 Additional Loss Coefficient {W/K}",
+        "  3,                       !- Node 3 Additional Loss Coefficient {W/K}",
+        "  4,                       !- Node 4 Additional Loss Coefficient {W/K}",
+        "  5,                       !- Node 5 Additional Loss Coefficient {W/K}",
+        "  6,                       !- Node 6 Additional Loss Coefficient {W/K}",
+        "  7,                       !- Node 7 Additional Loss Coefficient {W/K}",
+        "  ,                        !- Node 8 Additional Loss Coefficient {W/K}",
+        "  ,                        !- Node 9 Additional Loss Coefficient {W/K}",
+        "  ;                        !- Node 10 Additional Loss Coefficient {W/K}",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool ErrorsFound = false;
+    HeatBalanceManager::GetZoneData(ErrorsFound); // read zone data
+    EXPECT_FALSE(ErrorsFound);
+
+    InternalHeatGains::GetInternalHeatGainsInput(state);
+    EXPECT_FALSE(WaterThermalTanks::GetWaterThermalTankInput(state));
+
+    EXPECT_EQ(1, WaterThermalTanks::numWaterHeaterStratified);
+    WaterThermalTanks::WaterThermalTankData &Tank = WaterThermalTanks::WaterThermalTank(1);
+    EXPECT_EQ(6, Tank.Nodes);
+    EXPECT_EQ(6, Tank.AdditionalLossCoeff.size());
+    for (int i = 1; i <= Tank.Nodes; ++i) {
+        if (i==2) {
+            EXPECT_EQ(0, Tank.AdditionalLossCoeff(i));
+        } else {
+            EXPECT_EQ(i, Tank.AdditionalLossCoeff(i));
+        }
+    }
+
+
+    std::string const error_string = delimited_string({
+        "   ** Warning ** WaterHeater:Stratified = STRATIFIED TANK:  More Additional Loss Coefficients were entered than the number of nodes; extra coefficients will not be used",
+    });
+
+    EXPECT_TRUE(compare_err_stream(error_string));
```

Comment:
>     * has warning message "comparison between signed and unsigned integer expressions [-Wsign-compare]" that needs to be addressed. I suspect this could be caused by [this](https://github.com/NREL/EnergyPlus/blob/8163_WaterHeaterStratified_Nodes_Warning/tst/EnergyPlus/unit/WaterThermalTanks.unit.cc#L5103-L5105) due to integer and real comparison in each of three unit tests added.

Good catch, didn't see that warning locally, thanks!

For root cause: Real & int comparison isn't a problem. Array.size() is the problem:

```diff
-    EXPECT_EQ(6, Tank.AdditionalLossCoeff.size());
+   EXPECT_EQ(6u, Tank.AdditionalLossCoeff.size());
```

Reply:
Should be fixed by https://github.com/NREL/EnergyPlus/pull/8172/commits/55e4fd8793a6838455b1b4eddce716582eafd169 hopefully.